### ISSUE TITLE
jx quickstart with On-Prem Github Enterprise Fails

### ISSUE THREAD

**Issue by chaseconey** [jenkins-x/jx#511]
It seems like the `jx create quickstart` does not work when trying to target a different git server that is not github.com. I have added the node-http quickstart to our org and still can't seem to get it to pick up.

**Command Run:** `jx create quickstart -g other-org`

**Expected Result:** node-http quickstart to show up

**Actual Result:** error: No quickstarts match filter

**Version Output:**

```
jx             1.1.32
Jenkins X      0.0.393
Kubernetes     v1.8.9
Helm Client    v2.8.1+g6af75a8
Helm Server    v2.8.1+g6af75a8
Kubectl Client v1.9.5
Git            git version 2.16.2
```

**Comment by rawlingsj** [comment]
Thanks for reporting, I think the current logic wont prompt for new git server auth details if it doesn't match the original git server that was used in the install.

Yes it definitely is a bug and one we should hopefully be able to fix very soon.

I you wanted to try something in the meantime I wonder if you edit your `~/.jx/gitAuth.yaml` file and add another `server` section, easier to copy an existing one assuming github was originally used during the install.

Then add the details of your github enterprise server, e.g.
```
- url: my.github.server.io
  users:
  - username: [login user]
    apitoken: [a personal access token]
  name: GitHubEnterprise
  kind: GitHub
  currentuser: [login user]
```

That may work but we'll try and get this fixed up this week.

**Comment by chaseconey** [comment]
@rawlingsj Yeah, I was mucking around with that and it does seem to give the same result as reported. And in fact even when I tell it to search github.com rather than my github enterprise endpoint for quickstarts, that fails as well if that makes sense.

**Comment by rawlingsj** [comment]
OK thanks for trying, will try and get a fix sorted.

**Comment by jstrachan** [comment]
Btw I added some docs on adding other git providers
http://jenkins-x.io/developing/git/

Am guessing the URL for your GHE service will need to use `https://` or `http://`

We’ve not had a chance to test GHE yet - am hoping we can get that going pretty quickly

**Comment by jstrachan** [comment]
@chaseconey btw if you want to create the go QuickStart and try host the code on GHE maybe try add your GHE service (in the docs in the above comment) then try just use 

    jx create quickstart

As the `-g` option is for searching other organizations for QuickStarts to create - rather than specifying where to put the generated app

**Comment by jstrachan** [comment]
I've done lots of testing of installing Jenkins X using only GHE or creating apps/environments on GHE and I think this is fixed now with `0.0.699` of the jenkins-x-platform
